### ISSUE TITLE
[BUG] urllib3 not found when use it with pyinstaller and docker

### ISSUE THREAD

**Issue by yingshaoxo** [python-telegram-bot/python-telegram-bot#1877]
### Steps to reproduce
1. clone the repository: https://github.com/yingshaoxo/TGAsk.git

2. run pyinstaller: `pyinstaller main.py -w`

3. build a docker image: `sudo docker build -t yingshaoxo/tgask .`

4. add your `token` to `docker-compose.yml`

5. run docker-compose: `sudo docker-compose up`
### Expected behaviour
The program should run well as it would be outside of the docker container.

### Actual behaviour
The error message I got:
```
yingshaoxo@pop-os:~/Software_Engineering$ sudo docker-compose up 
Recreating software_engineering_TGAsk_1 ... done
Attaching to software_engineering_TGAsk_1
TGAsk_1  | [6] Failed to execute script main
TGAsk_1  | Traceback (most recent call last):
TGAsk_1  |   File "telegram/vendor/ptb_urllib3/urllib3/util/ssl_.py", line 308, in ssl_wrap_socket
TGAsk_1  | FileNotFoundError: [Errno 2] No such file or directory
TGAsk_1  | 
TGAsk_1  | During handling of the above exception, another exception occurred:
TGAsk_1  | 
TGAsk_1  | Traceback (most recent call last):
TGAsk_1  |   File "telegram/utils/request.py", line 225, in _request_wrapper
TGAsk_1  |   File "telegram/vendor/ptb_urllib3/urllib3/request.py", line 66, in request
TGAsk_1  |   File "telegram/vendor/ptb_urllib3/urllib3/request.py", line 87, in request_encode_url
TGAsk_1  |   File "telegram/vendor/ptb_urllib3/urllib3/poolmanager.py", line 244, in urlopen
TGAsk_1  |   File "telegram/vendor/ptb_urllib3/urllib3/connectionpool.py", line 617, in urlopen
TGAsk_1  |   File "telegram/vendor/ptb_urllib3/urllib3/connectionpool.py", line 360, in _make_request
TGAsk_1  |   File "telegram/vendor/ptb_urllib3/urllib3/connectionpool.py", line 857, in _validate_conn
TGAsk_1  |   File "telegram/vendor/ptb_urllib3/urllib3/connectionpool.py", line 289, in _validate_conn
TGAsk_1  |   File "telegram/vendor/ptb_urllib3/urllib3/connection.py", line 326, in connect
TGAsk_1  |   File "telegram/vendor/ptb_urllib3/urllib3/util/ssl_.py", line 310, in ssl_wrap_socket
TGAsk_1  | telegram.vendor.ptb_urllib3.urllib3.exceptions.SSLError: [Errno 2] No such file or directory
TGAsk_1  | 
TGAsk_1  | During handling of the above exception, another exception occurred:
TGAsk_1  | 
TGAsk_1  | Traceback (most recent call last):
TGAsk_1  |   File "main.py", line 84, in <module>
TGAsk_1  |   File "telegram/ext/updater.py", line 260, in start_polling
TGAsk_1  |   File "telegram/ext/jobqueue.py", line 311, in start
TGAsk_1  |   File "telegram/bot.py", line 54, in decorator
TGAsk_1  |   File "<decorator-gen-1>", line 2, in get_me
TGAsk_1  |   File "telegram/bot.py", line 67, in decorator
TGAsk_1  |   File "telegram/bot.py", line 279, in get_me
TGAsk_1  |   File "telegram/utils/request.py", line 277, in get
TGAsk_1  |   File "telegram/utils/request.py", line 231, in _request_wrapper
TGAsk_1  | telegram.error.NetworkError: urllib3 HTTPError [Errno 2] No such file or directory
```

### Configuration
**Operating System:** Linux pop-os 5.0.0-21-generic #22+system76-Ubuntu SMP Tue Jul 16 19:57:52 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

It's actually `Ubuntu 19.10`

**Version of Python, python-telegram-bot & dependencies:**
```
python-telegram-bot 12.5.1
certifi 2018.08.24
future 0.18.2
Python 3.7.3 (default, Oct  7 2019, 12:56:13)  [GCC 8.3.0]
```

**Comment by tsnoam** [comment]
This library is available via pypi to be installed using the `pip` utility.
We do not aim to support any other packaging mechanism.

That said, any contribution assisting to support other mechanisms is welcome.

**Comment by yingshaoxo** [comment]
I finally managed to do that by using the following command:
```
FROM ubuntu:19.10

RUN apt-get update \
  && apt-get install -y python3-pip python3-dev \
  && cd /usr/local/bin \
  && ln -s /usr/bin/python3 python \
  && pip3 install --upgrade pip

RUN pip3 install python-telegram-bot

COPY ./dist/main /software

CMD ["/software/main"]
```

___

But by using pip, the drawback is obvious, the size of the container would go up to **500MB+**.

Meanwhile, a normal pyinstaller container would only take **100MB** per project generally.

**Comment by tsnoam** [comment]
@yingshaoxo I am not sure how pyinstaller works. 
However from the logs I would suspect that they&#x27;re not cloning the git repo recursively.

**Comment by Poolitzer** [comment]
@yingshaoxo Random thought: If urrlib3 is the only issue, we allow you to have it installed on your own and not use our bundled version. Maybe that works for you?

**Comment by yingshaoxo** [comment]
@Poolitzer 

Hi! I appreciate your thoughts. You are right. 

I just did an experiment. 

I found that if I only install the `urllib3`, by using pyinstaller, it still work:
```
FROM ubuntu:19.10

RUN apt-get update \
  && apt-get install -y python3-pip python3-dev \
  && cd /usr/local/bin \
  && ln -s /usr/bin/python3 python \
  && pip3 install --upgrade pip

#RUN pip3 install python-telegram-bot  #attension!!!!!!!!! I comment out the installation of python-telegram-bot
RUN pip3 install urllib3

COPY ./dist/main /software

CMD ["/software/main"]
```

____


Yes, I don't think it's necessary to include your own bundled version of `urllib3`. **It makes pyinstaller confused.**

**Comment by Poolitzer** [comment]
@yingshaoxo Oh it is necessary, since we changed some lines in there which improved the experience of developers using the library, and urrlib3 didn&#x27;t accept our PR back then. And it decreases our dependencies.


In the current setup, everyone wins. The go-to dev enjoyes our always working library. And if in some environments this bundled urrlib version is not wanted, you can just have your own and everyone is happy :)

**Comment by yingshaoxo** [comment]
Oh my god! Finally, I found a perfect solution for all of us!

After a lot of thinking, I found that the real problem that stops me from getting it right at docker is the `SSL`.

I should have a look at these two lines earlier:
```
File "telegram/vendor/ptb_urllib3/urllib3/util/ssl_.py", line 308, in ssl_wrap_socket
TGAsk_1  | FileNotFoundError: [Errno 2] No such file or directory
```

I did a check at the 308 line of `/usr/local/lib/python3.7/dist-packages/telegram/vendor/ptb_urllib3/urllib3/util/ssl_.py`. 

There's a function called ` context.load_verify_locations(ca_certs, ca_cert_dir)`.

I printed `ca_certs` out, and I found `/etc/ssl/certs/ca-certificates.crt`.

I did some searches on Google. I doubt that this error may have related to `ca-certificates`.

____

So in the end, I just added the following code to my `Dockerfile`:
```
RUN apt-get update
RUN apt-get install -y ca-certificates
RUN update-ca-certificates
```

And it works !!! 
Now my docker image size goes down from `480MB` to `122MB`. Big win, isn't it?
